pkg/services/ngalert/writer/prom.go

```diff
@@ -38,17 +58,14 @@ func PointsFromFrames(name string, t time.Time, frames data.Frames, extraLabels
 
 	points := make([]Point, 0, len(col.Refs))
 	for _, ref := range col.Refs {
-		var f float64
-		if fp, empty, err := ref.NullableFloat64Value(); !empty && fp != nil {
+		// Use a default value of NaN if the value is empty or nil.
+		f := math.NaN()
```

Comment:
Discussed offline, we may rethink this in a future PR

Reply:
More concretely, writing NaN may not be desirable in HA setups. Instead we may just want to fail evaluation on that peer.